### ISSUE TITLE
Window resizing is broken on windows desktop.

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#134262]
No error messages printed, run the flutter gallery and resize. I'd expect the application to re-render at least after resizing finishes. Instead the entire screen goes black

This happens on ToT.

**Comment by jonahwilliams** [comment]
@cbracken @loic-sharma

**Comment by TahaTesser** [comment]
> No error messages printed, run the flutter gallery and resize. I'd expect the application to re-render at least after resizing finishes. Instead the entire screen goes black

It happens on Windows when switching from one display to another, which I think also resizes the window.

**Comment by sirpengi** [comment]
I'm having this same issue on windows on `main` but not in `beta` (`3.14.0-0.2.pre`)

**Comment by sirpengi** [comment]
I've run `git-bisect` and the issue shows up in da51c6c7183937b6d8f4f013f85eac047d414a6d

**Comment by gnprice** [comment]
> I've run `git-bisect` and the issue shows up in [da51c6c](https://github.com/flutter/flutter/commit/da51c6c7183937b6d8f4f013f85eac047d414a6d)

Looks like that was an engine roll:

https://github.com/flutter/engine/compare/489c399e3dc6...e496eec40e21

2023-09-02 skia-flutter-autoroll@skia.org Roll Skia from 2d8849f9f0cc to 15f77147a3ec (1 revision) (https://github.com/flutter/engine/pull/45414)
2023-09-02 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from OF4TS05qlWCjukWw6... to MesZPNdj-uw8VdCyV... (https://github.com/flutter/engine/pull/45413)
2023-09-02 dkwingsmt@users.noreply.github.com Remove --disable-service-auth-codes (https://github.com/flutter/engine/pull/45356)
2023-09-02 bdero@google.com [Impeller] Import cstring for memcpy. (https://github.com/flutter/engine/pull/45408)
2023-09-02 skia-flutter-autoroll@skia.org Roll Dart SDK from cdf1ce0c6d7e to a5c7102af509 (1 revision) (https://github.com/flutter/engine/pull/45412)
2023-09-02 skia-flutter-autoroll@skia.org Roll ANGLE from 179bd7762ffa to ebf1e7163216 (1 revision) (https://github.com/flutter/engine/pull/45411)
2023-09-02 dkwingsmt@users.noreply.github.com Remove deprecated MOCK_METHODx calls (https://github.com/flutter/engine/pull/45307)
2023-09-02 jonahwilliams@google.com [Impeller] Better demonstrate blur and draw picture? (https://github.com/flutter/engine/pull/45388)
2023-09-02 jonahwilliams@google.com [Impeller] Make paths externally immutable, update all tests to use PathBuilder to create Path. (https://github.com/flutter/engine/pull/45393)
2023-09-02 skia-flutter-autoroll@skia.org Roll ANGLE from 962fdf7b7882 to 179bd7762ffa (1 revision) (https://github.com/flutter/engine/pull/45409)
2023-09-02 flar@google.com Cull the RTree bounds when they are forwarded in DrawDisplayList (https://github.com/flutter/engine/pull/45358)
2023-09-02 skia-flutter-autoroll@skia.org Roll Skia from fedff79a6afc to 2d8849f9f0cc (3 revisions) (https://github.com/flutter/engine/pull/45407)
2023-09-02 jonahwilliams@google.com [impeller] premultiply vertices colors. (https://github.com/flutter/engine/pull/45406)
2023-09-01 skia-flutter-autoroll@skia.org Roll ANGLE from 6a09e41ce6ea to 962fdf7b7882 (224 revisions) (https://github.com/flutter/engine/pull/45400)
2023-09-01 skia-flutter-autoroll@skia.org Roll Skia from 22ae23891e8e to fedff79a6afc (1 revision) (https://github.com/flutter/engine/pull/45405)
2023-09-01 30870216+gaaclarke@users.noreply.github.com [Impeller] turned on validations for all debug builds (https://github.com/flutter/engine/pull/45350)
2023-09-01 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from sk7JBGzW1Jw10Wy-T... to OF4TS05qlWCjukWw6... (https://github.com/flutter/engine/pull/45403)
2023-09-01 skia-flutter-autoroll@skia.org Roll Skia from 2c0405489966 to 22ae23891e8e (1 revision) (https://github.com/flutter/engine/pull/45402)
2023-09-01 737941+loic-sharma@users.noreply.github.com [Windows] Update vsync on raster thread (https://github.com/flutter/engine/pull/45310)
2023-09-01 skia-flutter-autoroll@skia.org Roll Dart SDK from a2ea759c16cc to cdf1ce0c6d7e (1 revision) (https://github.com/flutter/engine/pull/45397)
2023-09-01 skia-flutter-autoroll@skia.org Roll Skia from f3f6c733c7e6 to 2c0405489966 (1 revision) (https://github.com/flutter/engine/pull/45396)
2023-09-01 skia-flutter-autoroll@skia.org Roll Skia from 02fa14799c6c to f3f6c733c7e6 (1 revision) (https://github.com/flutter/engine/pull/45394)
2023-09-01 skia-flutter-autoroll@skia.org Roll Skia from d5d3b0d4ee77 to 02fa14799c6c (2 revisions) (https://github.com/flutter/engine/pull/45392)
2023-09-01 41930132+hellohuanlin@users.noreply.github.com [ios][ios17][text_input]fix text input system highlight in iOS 17 Beta 7 with firstRectForRange (https://github.com/flutter/engine/pull/45303)
2023-09-01 skia-flutter-autoroll@skia.org Roll Skia from d6266ef14a7e to d5d3b0d4ee77 (2 revisions) (https://github.com/flutter/engine/pull/45389)
2023-09-01 skia-flutter-autoroll@skia.org Roll Dart SDK from 0c121a6431cc to a2ea759c16cc (1 revision) (https://github.com/flutter/engine/pull/45384)

Also rolling transitive DEPS:
  fuchsia/sdk/core/mac-amd64 from sk7JBGzW1Jw1 to MesZPNdj-uw8

**Comment by cbracken** [comment]
Of these https://github.com/flutter/engine/pull/45310 _seems_ like the most likely candidate (it was required in order to unblock the ANGLE roller)

**Comment by xu-baolin** [comment]
I also encountered this issue.

**Comment by kekland** [comment]
Can reproduce on a blank app:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blueGrey,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Text('hi'),
      ),
    );
  }
}
```

Doctor output:
```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel beta, 3.15.0-15.1.pre, on Microsoft Windows [Version 10.0.19045.3448], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    X cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.7.3)
[√] Android Studio (version 2022.3)
[√] VS Code (version 1.82.0)
[√] Connected device (3 available)
[√] Network resources
```

Nothing printed in the console.

https://github.com/flutter/flutter/assets/14993994/93f4f90b-4970-4bff-ac72-7ed8b7b09eaa

**Comment by wanghongenpin** [comment]
This problem has a serious impact. When will it be fixed?

**Comment by cbracken** [comment]
Engine patch merged in flutter/engine#46000. Should be resolved once the engine rolls to the framework.

**Comment by cbracken** [comment]
This rolled to master branch of the framework about 7 hours ago. If you sync the framework past 8936504, you'll have the fix.
